packages/meditrak-server/src/routes/DeleteHandler/DeleteHandler.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import {
+  constructRecordExistsWithId,
+  DatabaseError,
+  respond,
+  singularise,
+  ValidationError,
+} from '@tupaia/utils';
+import { CRUDHandler } from '../CRUDHandler';
+
+/**
+ * Responds to DELETE requests for a resource
+ */
+
+export class DeleteHandler extends CRUDHandler {
+  constructor(req, res) {
+    super(req, res);
+    this.model = this.models[singularise(this.resource)];
+  }
+
+  async handleRequest() {
+    await this.validate();
+    await this.deleteRecord();
+    respond(this.res, { message: `Successfully deleted ${this.resource}` });
+  }
+
+  async validate() {
+    if (!this.model.isDeletableViaApi) {
+      throw new ValidationError(`${this.resource} is not a valid DELETE endpoint`);
+    }
```

Comment:
We can delete this, and get rid of `isDeletableViaApi` now that we're listing them explicitly. It's the weird delete equivalent of `EDITABLE_RECORD_TYPES` in `editRecord.js`.

If deleting `isDeletableViaApi` means there is nothing of substance left in the meditrak-server version of the model, and it was extending from a model in `@tupaia/database`, we can also delete the model entirely from meditrak-server.

Reply:
Just something to note - Removing this check leaves our `validate` function with just a check for whether the entity exists. I think realistically most of the endpoints will also check that condition in the permissions gate, but worth keeping in the generic layer since conceptually that should be a generic test and "all delete handlers check this during the permissions gate" isn't necessarily always a valid assumption in our design.

Reply:
Good note, and agreed :+1: